lib/tests/test-with-nix.nix

```
@@ -32,7 +32,7 @@ pkgs.runCommand "nixpkgs-lib-tests-nix-${nix.version}" {
   nativeBuildInputs = [
     nix
     pkgs.gitMinimal
-  ] ++ lib.optional pkgs.stdenv.isLinux pkgs.inotify-tools;
+  ] ++ lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.inotify-tools;
```

```suggestion
  ] ++ lib.optional pkgs.stdenv.buildPlatform.isLinux pkgs.inotify-tools;
```

This probably applies to most conditionals for any `nativeBuildInputs`.

---

Let’s maybe fix these patterns in another PR, as it doesn’t match the current behaviour and would disrupt the primary metric (almost no rebuilds) we can judge this PR’s safety by? At least now it will be obviously wrong.

---

Good point, but maybe leave the bad ones as is, so they're easy to update in the follow-up PR? Or I guess this PR's diff could be used for that purpose. It's a bit more hidden, but about as good.

SGTM

---

I think it would require a smarter script than a simple text substitution to diagnose and avoid these cases, so I’d personally prefer to avoid holding this up over that.

---

`hostPlatform` may just as easily be correct for `nativeBuildInputs`. I'd just mechanistically replace them. Mistakes are mistakes before *and* after, they are just more obvious after.

I'd guess in most cases it also sort of doesn't matter since the derivations don't cross compile to begin with or (as is the case here) don't really make sense to cross compile.